### ISSUE TITLE
Feature Backlog

### ISSUE THREAD

**Issue by kevinpschaaf** [PolymerElements/iron-list#10]
- [x] Automatically adapting physicalItem count  - **p1** (Released in v1.0.1)
- [x] API to update the size of an item/subset of items  - **p1** (Released in v1.1.3)
- [x] Selection (single & multi) - **p1** (Released in v1.1.0)
- [ ] Grouping + dividers - **p1**
- [x] Grid - (Released in v1.3.0)
- [x] External scrollTarget  (it currently works for paper-scroll-header-panel)  (Released in v1.2.0)
  - Listen & measure on parent div
  - Measure offsetTop & take into account
- [x] Non-native scroller integration (Released in v1.2.0)
- [X] Accessibility (Released in v1.2.0)
- [ ] Polymorphic templates
- [ ] Swipable action sheets
- [ ] Bottom-up mode
- [ ] Horizontal mode
- [ ] Drag-and-drop reordering

**Comment by moderndeveloperllc** [comment]
:+1: for External scrollTarget and selection. Pretty much the last things that are keeping me from moving to 1 from .5.

**Comment by Pickachu** [comment]
:+1: for selection

**Comment by pdf** [comment]
Before this gets out of hand with people spamming +1 for every feature on this list, there's a subscribe button over there ->, use that to get notified of updates rather than spamming everyone who is subscribed.  We all have features we need from this plugin, that's why there's a list of them in the issue description.

**Comment by nomego** [comment]
:+1: for grouping

**Comment by gertcuykens** [comment]
Is the flex height some where in the backlog because I can't fit the screen? http://jsbin.com/qiyifet/edit?html,output

**Comment by birkir** [comment]
+1 for the grid

**Comment by indolering** [comment]
**DON'T** POST '+1' or :+1:, USE THE SUBSCRIBE BUTTON!!!

@pdf: Gotta make it obvious : (

**Comment by knyto2** [comment]
Is there a feature where the columns are fixed? This is for a scenario where there maybe a lot of columns and you want the first one or two columns on the left fixed.

**Comment by robrez** [comment]
@knyto2 You may want to check out vaadin's "v-grid"...
https://github.com/vaadin/components

It was highlighted during this year's IO Polymer talk...
https://youtu.be/fD2As5RmM8Q?t=1027

**Comment by RonnChyran** [comment]
What does p1 and p2 mean?

**Comment by blasten** [comment]
@RonnChyran the priorities :)

**Comment by agozie** [comment]
@robrez from my experience, vaadin v-grid is becomes very laggy with a dataset of a few hundred entries. But the iron-list is butter smooth, no matter what i threw at it. I had to replace vaadin v-grid with the iron-list, though with some effort that I don't regret.

**Comment by RonnChyran** [comment]
@blasten That's too bad then seeing as my app relies heavily on grid :/

**Comment by jojule** [comment]
@agozie would would you give some more detail on v-grid use? It would help us to fix the problem.

**Comment by jounik** [comment]
@agozie: Indeed, a simple test case would be helpful, and details about your environment.

We’re actually doing performance profiling at the moment, trying to identify the bottlenecks.

**Comment by gertcuykens** [comment]
PS Mine is still stuk at 20 max using iron-list.git#1.0.1 ?

**Comment by agozie** [comment]
@jojule @gertcuykens 
 ```javascript req.success(json.data, json.number);``` once json.data is an empty array, the grid fails to load subsequent data. My application involves displaying table data for clients, some have data, others empty. Once a client without data is encountered, ```javascript this.$.grid.data.clearCache() ``` fails to load data on subsequent clients. This is my main reason for dropping vaadin.

**Comment by jojule** [comment]
@agozie Sounds like a memory leak, not a performance problem. More details would be needed to research the issue.

To avoid hijacking the thread for things unrelated to iron list, would you be so kind to email to
jouni@vaadin.com about details?

**Comment by shanika** [comment]
Any idea when  Grouping + dividers will be available? Thanks

**Comment by gertcuykens** [comment]
For me selection is only 50% completed because you can't preselect out of the box. See http://stackoverflow.com/questions/32297052/iron-list-keep-selection-highlight-on-refresh

**Comment by davidrissato** [comment]
Please, could you put "Horizontal mode" in the priority list? I hope it supports "horizontal wrap" flow.

**Comment by Wurper** [comment]
Im also very interested in any information regarding the progress on the Grouping + dividers - feature.

**Comment by GrahenKraken** [comment]
Is there any way to get an approximate ETA for "Grid"? I know I'm probably asking too much, but my app's performance hinges on it.

**Comment by rbjarnason** [comment]
+1 on Grid support :)

**Comment by nomego** [comment]
@kevinpschaaf Is there any spec for the grouping support, anything I can help out with? It's the only showstopper for us migrating our whole application to Polymer 1.0.

**Comment by Andersmholmgren** [comment]
+100 on the drag n drop reordering

**Comment by myuseringithub** [comment]
Excellent job, thank you so much !

**Comment by Saulis** [comment]
Hate to be spamming like this, but in case someone is interested, I've been extending `iron-list` towards being a data grid: http://saulis.github.io/iron-data-table/

Comments, feedback, help – all would be highly appreciated.

**Comment by nomego** [comment]
I'll take the opportunity to present `cosmoz-grouped-list` - an approach to a layer component to support different templates for group headers and group items in `iron-list`: https://github.com/Neovici/cosmoz-grouped-list

This is a component central to our `cosmoz-omnitable` component: https://github.com/Neovici/cosmoz-omnitable

**Comment by redaready** [comment]
+100 Horizontal mode

**Comment by bobabellon** [comment]
is the backlog checklist updated updated? not sure if horizontal mode is implemented yet, updated to latest version of iron-list but i dont see horizontal mode in it...

**Comment by davidrissato** [comment]
I wrote a small article in 2015 that says how you can achieve horizontal grid columns using iron-list:

[https://easyusedev.wordpress.com/2015/10/02/polymer-how-to-iron-list-horizontal-grid/](https://easyusedev.wordpress.com/2015/10/02/polymer-how-to-iron-list-horizontal-grid/)

Since iron-list is not under active development, that approach might be helpful.

**Comment by bobabellon** [comment]
@davidrisatto would that work for an infinite horizontal list? Or only if you have a definite or pre defined number of columns?

**Comment by davidrissato** [comment]
@bababellon No, that solution is more towards a vertical grid with multiple horizontal cells per line.
